ui-core/src/components/inputs/InputWithSuggestions.tsx

```diff
@@ -0,0 +1,130 @@
+  numberOfSuggestionsToShow?: number;
+  exactSearch?: boolean;
+}
+
+const InputWithSuggestions = <T,>({
+  suggestions,
+  onChange,
+  getSuggestionLabel,
+  customLabel,
+  numberOfSuggestionsToShow = 5,
+  exactSearch = false,
+  ...inputProps
+}: InputWithSuggestionsProps<T>) => {
+  const [filteredSuggestions, setFilteredSuggestions] = useState<T[]>([]);
+  const [activeSuggestion, setActiveSuggestion] = useState(-1);
+  const [showSuggestions, setShowSuggestions] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleInputChange: ChangeEventHandler<HTMLInputElement> = (e) => {
+    const userInput = e.currentTarget.value.toLowerCase();
+
+    if (userInput === '') {
+      setFilteredSuggestions([]);
+      setShowSuggestions(false);
+      return;
+    }
+
+    const filtered = suggestions.filter((suggestion) => {
+      const suggestionLabel = getSuggestionLabel(suggestion).toLowerCase();
+      return exactSearch
+        ? suggestionLabel.startsWith(userInput)
+        : suggestionLabel.includes(userInput);
+    });
+
+    setFilteredSuggestions(filtered);
+    setShowSuggestions(true);
+  };
+
+  const handleKeyDown: KeyboardEventHandler<HTMLInputElement> = (e) => {
+    if (e.key === 'ArrowDown') {
+      setActiveSuggestion((prev) => (prev < filteredSuggestions.length - 1 ? prev + 1 : prev));
+    } else if (e.key === 'ArrowUp') {
+      setActiveSuggestion((prev) => (prev > 0 ? prev - 1 : prev));
+    } else if ((e.key === 'Enter' || e.key === 'Tab') && activeSuggestion >= 0) {
+      selectSuggestion(activeSuggestion);
+    } else if (e.key === 'Escape') {
+      setShowSuggestions(false);
+      setFilteredSuggestions([]);
+    }
+  };
+
+  const selectSuggestion = (index: number) => {
+    const selectedSuggestion = filteredSuggestions[index];
+    if (inputRef.current) {
+      inputRef.current.value = getSuggestionLabel(selectedSuggestion);
+    }
+    onChange(selectedSuggestion);
+    setFilteredSuggestions([]);
+    setShowSuggestions(false);
+  };
+
+  useEffect(() => {
+    if (showSuggestions && filteredSuggestions.length === 0) {
+      setShowSuggestions(false);
+    }
+  }, [filteredSuggestions, showSuggestions]);
+
+  const handleInputFocus: FocusEventHandler<HTMLInputElement> = () => {
+    if (inputRef.current && inputRef.current.value === '') {
+      setFilteredSuggestions(suggestions);
+      setShowSuggestions(true);
+    }
+  };
+
+  return (
+    <div
+      className="input-with-suggestions"
+      style={{ '--number-of-suggestions': numberOfSuggestionsToShow } as React.CSSProperties}
+    >
+      {customLabel && <label htmlFor={inputProps.id}>{customLabel}</label>}
+      <Input
+        {...inputProps}
+        ref={inputRef}
+        onChange={handleInputChange}
+        onKeyDown={handleKeyDown}
+        onBlur={() => {
+          setTimeout(() => setShowSuggestions(false), 200);
+        }}
+        onFocus={handleInputFocus}
+        withChevronButton
```

Comment:
I still think we should show the chevron icon only if there is suggestions provided. For example, when using this component for the stdcm form where we call the search endpoint to have suggestions displayed, the user might be tempted to click on the chevron to display something but it won't if he didn't type anything in the input.

```suggestion
        withChevronButton={suggestions.length > 0}
```

Reply:
je pense qu'il faut voir avec @thibautsailly

Reply:
Maybe, pass a prop to choose the desired behavior?

Reply:
i went with @SharglutDev suggestion, some visual enhancements may need @thibautsailly  approuval

Reply:
Thank you for the changes but my point was on the chevron icon, not the clear one.

In the stdcm sketch for the form, I'm guessing we will use these inputWithSuggestion to search for operational points.

![Capture d’écran 2024-08-08 à 11 43 48](https://github.com/user-attachments/assets/70ab8011-5b69-4f5a-9aee-bcbf35a76bd5)

But for these inputs, we won't pass anything in the `suggestions` prop as we won't have anything to pass if we don't type anything in the input. 
So the user shouldn't see the chevron icon when the input is empty (as we can see in the sketch).

So following change should be done : when `suggestions.length === 0`, we shouldn't see the chevron icon.

Reply:
Fixed